src/gamemode.c

```diff
@@ -280,6 +317,40 @@ handle_call_thread (GTask        *task,
   g_dbus_method_invocation_return_value (call->inv, g_variant_new ("(i)", r));
 }
 
+static void
+handle_call_in_thread_fds (XdpGameMode           *object,
+                           const char            *method,
+                           GDBusMethodInvocation *invocation)
+{
+  g_autoptr(GTask) task = NULL;
+  GDBusMessage *msg;
+  GUnixFDList *fdlist;
+  XdpAppInfo *app_info;
+  Request  *request;
+  CallData *call;
+
+  msg = g_dbus_method_invocation_get_message (invocation);
+  fdlist = g_dbus_message_get_unix_fd_list (msg);
+
+  if (fdlist == NULL || g_unix_fd_list_get_length (fdlist) != 2)
+    {
+      g_dbus_method_invocation_return_error (invocation, G_DBUS_ERROR, G_DBUS_ERROR_INVALID_ARGS,
+                                             "File descriptor number is incorrect");
+      return;
+    }
+
+  request = request_from_invocation (invocation);
+  app_info = request->app_info;
+
+  call = call_data_new (invocation, app_info, method);
+  call->fdlist = g_object_ref (fdlist);
```

Comment:
Here is a question: can we recognize if an fd is a pidfd ? Because the portal frontend is supposed to validate arguments before passing them on to the backend, so it would be nice to check here, and reject 'bad' fds

Reply:
That is a good point. We can actually do that, by checking that `/proc/self/fdinfo/<fd>` contains a `Pid: X` entry, and additionally that X is not zero. I will add that check.